### DISCUSSION TITLE
Optiga Trust M Arbitrary Data capacity

### DISCUSSION THREAD

**Discussion by DIEGGOMEZ** [Infineon/optiga-trust-m#116] [Q&A] [answered]
Hello,

I am working with Optiga Trust M V3, my doubt is about Arbitrary Data slots capacity, i have seen that 0xF1D0 – 0xF1DB slots are 1500 bytes and type 3, and F1E0 – 0xF1E1 are 140 bytes and type 2.

Doing some test i have realised that this slots are inverted, but i am not sure about the real answer.

Link where i found the info: https://community.infineon.com/t5/Knowledge-Base-Articles/OPTIGA-Trust-M-Write-Data-KBA235372/ta-p/354386

Any idea?

Regards.

**Comment by koblibri** [comment, marked as answer]
Hi,

you are correct: The OPTIGA Trust M has 12 "Type 3" Arbitrary Data Slots (0xF1D0 - 0xF1DB) which can store up to 140 Bytes each and 2 "Type 2" Slots (0xF1E0 & 0xF1E1), which can store up to 1500 Bytes each.
You can find this information inside the Solution Reference Manual, which is part of the [Documents](https://github.com/Infineon/optiga-trust-m/tree/develop/documents) folder.

**Comment by DIEGGOMEZ** [comment]
Perfect, and about the endurance of each slot? Type 2 (1500 bytes) are less of 100 cycles or they are often changed group?

Regards.

**Comment by koblibri** [comment]
Type 2 is of the group "Seldom Changed", which means that the NVM can only be programmed up to 100 times. 
![image](https://github.com/Infineon/optiga-trust-m/assets/57197288/f4c16d5f-4766-458d-b63c-6d01c7cc9010)

**Reply by DIEGGOMEZ** [reply]
Understood,

Thank you.

**Comment by sgsharath123** [comment]
@DIEGGOMEZ : We have now corrected the typo in the KBA
https://community.infineon.com/t5/Knowledge-Base-Articles/OPTIGA-Trust-M-Write-Data-KBA235372/ta-p/354386. 
Thanks for your feedback.

**Reply by DIEGGOMEZ** [reply]
No problem.

I have one last doubt, i have multiple PCBs populated with Optiga Trust M, in one hand some of them are able to open application and works correctly, but on the other hand some of them are unable to open application, report 0x102 error and i cant work properly with them.

It could be an hardware problem in these units?

Regards!

**Reply by DIEGGOMEZ** [reply]
@koblibri Any idea about this topic? 

Regards!

**Comment by sss551** [comment]
Hi @DIEGGOMEZ ,

The error code is the communication error between Optiga and host. Can you please check whether Optiga is powered up or not?

Best regards,
Shiva Sai
